apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DecimalReadTests.cs

```diff
@@ -0,0 +1,211 @@
+            // Write the decimal value and read it back.
+            await connection.CreateInsertOrUpdateCommand(_fixture.TableName,
+                new SpannerParameterCollection
+                {
+                    new SpannerParameter("DecimalValue", SpannerDbType.Float64, expectedValue),
+                }
+            ).ExecuteNonQueryAsync();
+            var decimalValue = await connection.CreateSelectCommand($"SELECT DecimalValue FROM {_fixture.TableName}").ExecuteScalarAsync<decimal>();
+            Assert.Equal(expectedValue, decimalValue);
+            var numericValue = await connection.CreateSelectCommand($"SELECT DecimalValue FROM {_fixture.TableName}").ExecuteScalarAsync<SpannerNumeric>();
+            Assert.Equal(SpannerNumeric.FromDecimal(expectedValue, LossOfPrecisionHandling.Truncate), numericValue);
+            // Explicitly test the scenario in which CLR Type is not specified.
+            var value = await connection.CreateSelectCommand($"SELECT DecimalValue FROM {_fixture.TableName}").ExecuteScalarAsync<object>();
+            Assert.True(value is SpannerNumeric);
+            Assert.Equal(SpannerNumeric.FromDecimal(expectedValue, LossOfPrecisionHandling.Truncate), (SpannerNumeric)value);
+        }
+
+        // Test Case: Using decimal with Float64 column in database without explicitly specifying SpannerDbType of parameter.
+        // SpannerConversionOptions is not explicitly set, so Default option would be used.
+        // Default type for decimal is Float64, so it will work with Float64 type column in database.
+        [Theory]
+        [MemberData(nameof(TestDecimals))]
+        public async Task WriteThenRead_ShouldBeEqual_WithoutSpannerDbType(decimal expectedValue)
+        {
+            using var connection = _fixture.GetConnection();
+            await connection.CreateInsertOrUpdateCommand(_fixture.TableName,
+                new SpannerParameterCollection
+                {
+                    // Without options, SpannerDbType of parameter will default to Float64.
+                    new SpannerParameter { ParameterName = "DecimalValue", Value = expectedValue },
+                }
+            ).ExecuteNonQueryAsync();
+            var decimalValue = await connection.CreateSelectCommand($"SELECT DecimalValue FROM {_fixture.TableName}").ExecuteScalarAsync<decimal>();
+            Assert.Equal(expectedValue, decimalValue);
+        }
+
+        // Test Case: Using decimal with Float64 column in database without passing SpannerDbType of the parameter.
+        // SpannerConversionOptions is set to use Numeric for decimal.
+        // This should throw SpannerException as Float64 column expects Float64 value but Numeric value is passed.
+        [Theory]
+        [MemberData(nameof(TestDecimals))]
+        public async Task WriteWithoutSpannerDbType_UseOptions(decimal expectedValue)
```

Comment:
I don't think this should be in DecimalReadTests.

(We can rename the test of course...)